### PR TITLE
CONN-1768-add capability to pass all subject confirmation method

### PR DESCRIPTION
**Pull request by mhpnguyen**
Include a new feature to pass all subject confirmation method (sender-vouches/bear) from nhinc layer to adapter.
@trantang (Optional), @pvenkatakrishnan1 (Optional)

### CODE REVIEW

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/saml/extraction/SamlTokenCreator.java [hunk 1]
```diff
@@ -58,7 +66,7 @@
      * @return A Map containing all of the information needed for creation of the SAML Token.
      */
     public Map<String, Object> createRequestContext(AssertionType assertion, String url, String action) {
-        LOG.debug("Entering SamlTokenCreator.CreateRequestContext...");
+        LOG.debug("Entering SamlTokenCreator.CreateRequestContext... version 1234");
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/saml/extraction/SamlTokenCreator.java]: Not sure what's the purpose of version 1234

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/saml/extraction/SamlTokenCreator.java [hunk 2]
```diff
@@ -309,4 +317,32 @@
...
+        for (SamlSubjectsType samlSubject : assertion.getSamlSubjects()){
+            String method = samlSubject.getMethod();
+            SubjectConfirmationDataBean bean = new SubjectConfirmationDataBean();
+            bean.setAddress(samlSubject.getAddress());
+            bean.setInResponseTo(samlSubject.getInResponseTo());
+            bean.setRecipient(samlSubject.getRecipient());
+            SamlConditionsType subjectCondition = samlSubject.getSubjectCondition();
+            if (subjectCondition != null) {
+                bean.setNotAfter(dateFormat.parseDateTime(subjectCondition.getNotOnOrAfter()));
+                bean.setNotBefore(dateFormat.parseDateTime(subjectCondition.getNotBefore()));
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/saml/extraction/SamlTokenCreator.java]: Check for null for both getNotOnOrAfter and getNotBefore before parsing

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/saml/extraction/SamlTokenCreator.java (hunk 1 above)
Comment (jsmith2) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/saml/extraction/SamlTokenCreator.java]: Remove version 1234.

Comment (jsmith2) [reply Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/saml/extraction/SamlTokenCreator.java]: Also, change this logging level to trace.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/saml/extraction/SamlTokenCreator.java [hunk 3]
```diff
@@ -309,4 +317,32 @@
...
+        DateTimeFormatter dateFormat = ISODateTimeFormat.dateTimeParser();
+        for (SamlSubjectsType samlSubject : assertion.getSamlSubjects()){
+            String method = samlSubject.getMethod();
+            SubjectConfirmationDataBean bean = new SubjectConfirmationDataBean();
+            bean.setAddress(samlSubject.getAddress());
+            bean.setInResponseTo(samlSubject.getInResponseTo());
+            bean.setRecipient(samlSubject.getRecipient());
+            SamlConditionsType subjectCondition = samlSubject.getSubjectCondition();
+            if (subjectCondition != null) {
+                bean.setNotAfter(dateFormat.parseDateTime(subjectCondition.getNotOnOrAfter()));
```
Comment (jsmith2) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/saml/extraction/SamlTokenCreator.java]: should we validate the dates or have the option to using the same property that we have for the other notbefore/notAfter values?

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/HOKSAMLAssertionBuilder.java [hunk 4]
```diff
@@ -203,12 +205,29 @@ protected Issuer createIssuer(final CallbackProperties properties, final X509Cer
...
+                SubjectConfirmation bearConfirmation = openSamlBuilder.createBearConfirmation(bear);
+                subject.getSubjectConfirmations().add(bearConfirmation);
+            }
+        }
+        // create voucher if exist
+        List<SubjectConfirmationDataBean> senderVouchesBeans = properties.getSenderVouchesBeans();
+        if (CollectionUtils.isNotEmpty(senderVouchesBeans)) {
+            for (SubjectConfirmationDataBean sender : senderVouchesBeans) {
+                SubjectConfirmation bearConfirmation = openSamlBuilder.createSenderVouchesConfirmation(sender);
+                subject.getSubjectConfirmations().add(bearConfirmation);
```
Comment (jsmith2) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/HOKSAMLAssertionBuilder.java]: Same variable names (change to senderVouches.

File: Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/callback/opensaml/HOKSAMLAssertionBuilderTest.java [hunk 5]
```diff
@@ -777,6 +780,18 @@ public String getNPI() {
             public String getUserOrganizationId() {
                 return "orgId";
             }
+
+            @Override
+            public List<SubjectConfirmationDataBean> getSenderVouchesBeans() {
+                // TODO Auto-generated method stub
```
Comment (jsmith2) [inline Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/callback/opensaml/HOKSAMLAssertionBuilderTest.java]: get rid of TODOs.

**Review by TabassumJafri — CHANGES_REQUESTED** [review]

**Review by TabassumJafri — APPROVED** [review]

**Review by jsmith2 — APPROVED** [review]

**Pull request merged**